src/NBitcoin/ChainedHeader.cs

```diff
@@ -4,6 +4,63 @@
 
 namespace NBitcoin
 {
+    /// <summary>
+    /// Represents the availability state of a block.
+    /// </summary>
+    public enum BlockDataAvailabilityState
+    {
+        /// <summary>
+        /// A <see cref="BlockHeader"/> is presented, the block data is not available.
+        /// </summary>
+        HeaderOnly,
+
+        /// <summary>
+        /// We are interested in downloading the <see cref="Block"/> that is being represented by the current <see cref="BlockHeader"/>. 
+        /// This happens when we don't have block which is represented by this header and the header is a part of a chain that
+        /// can potentially replace our consensus tip because its chain work is greater than our consensus tip's chain work.
+        /// </summary>
+        BlockRequired,
+
+        /// <summary>
+        /// The <see cref="Block"/> was downloaded and is available, but it may not be reachable directly but via a store.
+        /// </summary>
+        BlockAvailable
+    }
+
+    /// <summary>
+    /// Represents the validation state of a block.
+    /// </summary>
+    public enum ValidationState
+    {
+        /// <summary>
+        /// No validation was performed.
+        /// </summary>
+        Unknown,
+
+        /// <summary>
+        /// We have a valid block header.
+        /// </summary>
+        HeaderValidated,
+
+        /// <summary>
+        /// Blocks represented by headers with this state are assumed to be valid and only minimal validation is required for them when they are downloaded.
+        /// This state is used for blocks before the last checkpoint or for blocks that are on the chain of assume valid block.
+        /// Once a block is fully validated this state will change to <see cref="PartiallyValidated"/>.
+        /// </summary>
+        AssumedValid,
+
+        /// <summary>
+        /// Validated using all rules that don't require change of state.
+        /// </summary>
+        PartiallyValidated,
+
+        /// <summary>
+        /// Validated using all the rules.
+        /// Some rules validation may be skipped for blocks marked as <see cref="AssumedValid"/>. 
```

Comment:
add previously 
-> ``/// Some rules validation may be skipped for blocks previously marked as <see cref="AssumedValid"/>. ``